src/users/index.js

```diff
@@ -0,0 +1,217 @@
+   * @returns {PromiseLike<U | undefined> | U | undefined }
+   */
+  getUser (userId) { throw new Error('unimplemented') }
+
+  /**
+   * Fetches all the users who have claimed this task id.
+   *
+   * @param {string} taskId - The task id we want to query against
+   * @returns { AsyncGenerator<U> | Generator<U> | AsyncIterable<U> | Iterable<U> }
+   * @abstract
+   */
+  getClaimants (taskId) { throw new Error('unimplemented') }
+}
+
+/**
+ *
+ * @param {UserProvider<User, any>} userProvider a user provider
+ * @param { import("../config/index").Config["users"] } providerConfig the user provider configuration. See config/index.js for full schema
+ * @param { import("../config/index").Config } config the application configuration. See config/index.js for full schema
+ * @returns {Promise<express.Router>}
+ */
+async function mount (userProvider, providerConfig, config) {
+  const router = express.Router()
+  router.use(express.json())
+
+  // Route to handle `POST /users/:id/tasks/claim`.  This route is called when a
+  // user wants to user for a task.  It does not return any data, but
+  // should send a 201 HTTP status code to indicate that everything went OK.
+  router.post('/:id/tasks/claim', async (req, res) => {
+    try {
+      const { id } = req.params
+      const { tasks } = req.body
+
+      const user = await userProvider.getUser(id)
+
+      if (user) {
+        await user.claimTasks(tasks)
+        // Everything is OK.
+        res.status(201).end()
+      } else {
+        res.status(404).end()
+      }
+    } catch (error) {
+      logError(error)
+      res.status(500).json(error)
+    }
+  })
+
+  // Route to handle `POST /users/:id/tasks/claim`.  This route is called when a
+  // user wants to not user for a task.  It does not return any data, but
+  // should send a 201 HTTP status code to indicate that everything went OK.
+  router.post('/:id/tasks/unclaim', async (req, res) => {
+    try {
+      const { id } = req.params
+      const { tasks } = req.body
+
+      const user = await userProvider.getUser(id)
+
+      if (user) {
+        await user.unclaimTasks(tasks)
+        // Everything is OK.
+        res.status(201).end()
+      } else {
+        res.status(404).end()
+      }
+    } catch (error) {
+      logError(error)
+      res.status(500).json(error)
+    }
+  })
+
+  // Route to handle `GET /users/:id/`.
+  router.get('/:id/', async (req, res) => {
+    try {
+      const { id } = req.params
+
+      const user = await userProvider.getUser(id)
+
+      if (user) {
+        res.json({ id: user.id })
+      } else {
+        res.status(404).end()
+      }
+    } catch (error) {
+      logError(error)
+      res.status(500).json(error)
+    }
+  })
+
+  // Route to handle `GET /users/:id/tasks/`. This route will return all the tasks currently claimed by that user. It will return 404 if an invalid user is given
+  router.get('/:id/tasks/', async (req, res) => {
```

Comment:
I'm not sure we need methods to get any information about the user or their claimed tasks. I thought our approach to delivering this sort of information to Jerry would be by email or something that didn't require authentication?

Reply:
Yeah, I think I forgot that when I was working on this. That is a better approach